### PR TITLE
Add meta-mender support for the watchdog functionality in the C++ mender-client

### PR DESCRIPTION
**Pull request by oleorhagen**
I added some simple functionality for handling watchdog timeout support in meta-mender (configuring it).

Through the `MENDER_WATCHDOG_SEC` variable.

Also added a simple acceptance test for the watchdog functionality.

### CODE REVIEW

File: meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc [hunk 1]
```diff
@@ -344,6 +347,14 @@ do_install() {
     DESTDIR='${D}' cmake_runcmake_build --target install-dbus
 }
 
+do_install:append() {
+    sed -i 's/^[#[:space:]]*PasswordAuthentication.*/PasswordAuthentication no/' ${D}${sysconfdir}/ssh/sshd_config
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:351]: This is your own debugging code, isn't it? 🙂

File: tests/acceptance/test_watchdog.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        config["InventoryPollIntervalSeconds"] = 5
+        config["UpdatePollIntervalSeconds"] = 5
+        config["RetryPollIntervalSeconds"] = 2
+
+        with tempfile.NamedTemporaryFile() as mender_conf:
+            mender_conf.write(mender_conf_contents)
+            put_no_sftp(
+                mender_conf.name,
+                connection,
+                remote="/etc/mender/mender.conf",
```
Comment (kacf) [inline tests/acceptance/test_watchdog.py:132]: Isn't the image R/O by default? I think you can modify `/data/etc/mender/mender.conf` though (you should double check that path), which the path above points to.

Comment (kacf) [inline tests/acceptance/test_watchdog.py:132]: Client acceptance tests share QEMU instances, so the original file needs to be restored afterwards.

File: meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc [hunk 3]
```diff
@@ -14,6 +14,9 @@ MENDER_UPDATE_POLL_INTERVAL_SECONDS ?= "1800"
 MENDER_INVENTORY_POLL_INTERVAL_SECONDS ?= "28800"
 MENDER_RETRY_POLL_INTERVAL_SECONDS ?= "300"
 
+_MENDER_WATCHDOG_DEFAULT_TIMEOUT_SEC := 86400 # 24 hours
+MENDER_WATCHDOG_SEC ?= "${@max(d.getVar('MENDER_UPDATE_POLL_INTERVAL_SECONDS'), d.getVar('_MENDER_WATCHDOG_DEFAULT_TIMEOUT_SEC')}"
```
Comment (kacf) [inline meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:18]: This probably needs to be added to `meta-mender-core/conf/mender-vars.json`.

**Review by kacf — CHANGES_REQUESTED** [review]
Nice test

File: tests/acceptance/test_watchdog.py (hunk 2 above)
Comment (oleorhagen) [reply tests/acceptance/test_watchdog.py:132]: Hmm, I don't think they are? :thinking: 

This works at least

Comment (oleorhagen) [reply tests/acceptance/test_watchdog.py]: Nope, I just had a look at a few of the other tests. The `put_no_sftp` is pretty much copy paste from `test_update`

Comment (oleorhagen) [reply tests/acceptance/test_watchdog.py]: Hmm.. Seems ur right with the build conf. Then what is going on with the other tests?

**Review by kacf — APPROVED** [review]
Nice solution!

**Pull request merged**